pkg/linode-bs/controllerserver.go

```diff
@@ -265,11 +266,17 @@ func (linodeCS *LinodeControllerServer) ControllerPublishVolume(ctx context.Cont
 		return nil, status.Error(codes.Internal, err.Error())
 	}
 
-	opts := &linodego.VolumeAttachOptions{
-		LinodeID: linodeID,
-		ConfigID: 0,
+	// Check to see if there is room to attach this volume to the instance.
+	if canAttach, err := linodeCS.canAttach(ctx, linodeID); err != nil {
+		return &csi.ControllerPublishVolumeResponse{}, status.Error(codes.Internal, err.Error())
+	} else if !canAttach {
+		return &csi.ControllerPublishVolumeResponse{}, status.Error(codes.ResourceExhausted, "max number of volumes already attached to instance")
 	}
 
+	opts := &linodego.VolumeAttachOptions{
+		LinodeID:           linodeID,
+		PersistAcrossBoots: ptr.To(false),
+	}
 	if _, err := linodeCS.CloudProvider.AttachVolume(ctx, volumeID, opts); err != nil {
```

Comment:
can we really attach 64 volumes to one linode with not changing this call from the previous code?

Reply:
According to the API documentation, and various bits of implied information from other (internal) sources, it would really seem to be this simple.

Bear in mind, that these are preliminary changes, and I still have to do some thorough testing.